server/src/main/java/io/crate/execution/engine/aggregation/impl/TopKAggregation.java

```diff
@@ -198,13 +203,68 @@ public DataType<?> partialType() {
         return new StateType(boundSignature.argTypes().getFirst());
     }
 
+    @Nullable
+    @Override
+    public DocValueAggregator<?> getDocValueAggregator(LuceneReferenceResolver referenceResolver,
+                                                       List<Reference> aggregationReferences,
+                                                       DocTableInfo table,
+                                                       List<Literal<?>> optionalParams) {
+        if (!aggregationReferences.isEmpty()) {
+            Reference reference = aggregationReferences.getFirst();
+            if (reference.hasDocValues()) {
+                int limit;
+                if (optionalParams.size() == 1) {
+                    limit = (int) optionalParams.getFirst().value();
+                } else {
+                    limit = DEFAULT_LIMIT;
+                }
+                return getDocValueAggregator(reference, limit);
+            }
+        }
+        return null;
+    }
```

Comment:
Does this work for the case mentioned here: https://github.com/crate/crate/pull/16326#discussion_r1680620520 ?

Reply:
Yes, it works.

```
cr> select topk('1', 1);
+---------------------------------+
| topk('1', 1)                    |
+---------------------------------+
| [{"frequency": 1, "item": "1"}] |
+---------------------------------+
```

Reply:
In the case `topk('some-literal', int_column)` the method `getDocValueAggregator` is never called since there is not doc value available.

Reply:
That's a different example and it looks like it doesn't work:

```
cr> create table tbl (name text, x int);
CREATE OK, 1 row affected (0.562 sec)

cr> insert into tbl (name, x) values ('foo', 1);
INSERT OK, 1 row affected (0.052 sec)

cr> insert into tbl (name, x) values ('bar', 2);
INSERT OK, 1 row affected (0.028 sec)

cr> select topk(name, x) from tbl;
+--------------------------------------------------------------------+
| topk(name, x)                                                      |
+--------------------------------------------------------------------+
| [{"frequency": 1, "item": "bar"}, {"frequency": 1, "item": "foo"}] |
+--------------------------------------------------------------------+
SELECT 1 row in set (13.516 sec)

cr> select topk('asdf', x) from tbl;
ClassCastException[class java.lang.String cannot be cast to class java.lang.Integer (java.lang.String and java.lang.Integer are in module java.base of loader 'bootstrap')]

cr> \verbose
Verbose ON

cr> select topk('asdf', x) from tbl;
ClassCastException[class java.lang.String cannot be cast to class java.lang.Integer (java.lang.String and java.lang.Integer are in module java.base of loader 'bootstrap')]

java.lang.ClassCastException: class java.lang.String cannot be cast to class java.lang.Integer (java.lang.String and java.lang.Integer are in module java.base of loader 'bootstrap')
        at io.crate.execution.engine.aggregation.impl.TopKAggregation.getDocValueAggregator(TopKAggregation.java:217)
```


Passing a column to the second argument is of course generally a bit questionable as it will use the first value and ignore the rest, but it shouldn't fail like that.

Reply:
Ok, i did not have that on the radar. I got the case covered now.

```
cr> select topk('asdf', x) from tbl;
SQLParseException[Invalid parameters for topk aggregation.]
```

I cannot put the parameters back in the right in the feedback because i don't have the order anymore therefore. This would require to handle the original inputs into `getDocValueAggregator`.